ddtrace/_trace/tracer.py

```diff
@@ -1244,6 +1247,55 @@ def set_tags(self, tags: Dict[str, str]) -> None:
         """
         self._tags.update(tags)
 
+    def record_exception(
+        self,
+        exception: Exception,
+        attributes: Optional[Dict[str, str]] = None,
+        timestamp: Optional[int] = None,
+        escaped=False,
+    ) -> None:
+        """
+        Records an exception as span event.
+        If the exception is uncaught, :obj:`escaped` should be set :obj:`True`. It
+        will tag the span with an error tuple.
+
+        :param Exception exception: the exception to record<
+        :param dict attributes: optional attributes to add to the span event. It will override
+            the base attributes if :obj:`attributes` contains existing keys.
+        :param int timestamp: the timestamp of the span event. Will be set to now() if timestamp is :obj:`None`.
+        :param bool escaped: sets to :obj:`False` for a handled exception and :obj:`True` for a uncaught exception.
+        """
+        current_span = self.current_span()
+        if current_span is None:
+            return
+
+        if timestamp is None:
+            timestamp = time_ns()
+
+        exc_type, exc_val, exc_tb = type(exception), exception, exception.__traceback__
+
+        if escaped:
+            current_span.set_exc_info(exc_type, exc_val, exc_tb)
```

Comment:
nit: Here we convert trackbacks into a string twice (once in `set_exc_info`, once on line 1282). I know we do this in otel but it would be nice if we could consolidate the two. This doesn't block the PR. We can do this clean up as a follow up.

Reply:
that's a good idea too, i noted it.